benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp

```diff
@@ -0,0 +1,525 @@
+#include <simdjson.h>
+#include <vector>
+#include <limits>
+#include <random>
+#include <fstream>  // ofstream
+
+#include <cereal/archives/json.hpp>
+#include <cereal/cereal.hpp>
+#include <cereal/types/vector.hpp>
+
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+bool save_to_myfile(const std::string & str, const std::string & fileName)
+{
+    std::ofstream outFile;
+    outFile.open(fileName);
+    if(!outFile){
+        return false;
+    }
+    outFile << str << std::endl;
+    return true;
+}
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+float get_random_float_using_whole_range(){
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    return std::uniform_real_distribution<float>(std::numeric_limits<float>::max(), std::numeric_limits<float>::min())(gen);
+}
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// cereal needs to know which data members to serialize in your classes. 
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+struct myVectorFloats {
+    // As we use vector, we need to add related cereal include '#include <cereal/types/vector.hpp>'.
+    std::vector<float> _vec_f;
+
+    friend class cereal::access;
+
+    template<class Archive>
+    void serialize(Archive & archive) {
+        archive( cereal::make_nvp("vector" , _vec_f) );
+    }
+    
+    void serialize__Vector_to__JSON_file(const std::string & filename_string){
+        std::ofstream os(filename_string);
+        {
+            cereal::JSONOutputArchive archive_out(os);
+            serialize(archive_out);
+        }
+        return;
+    }
+
+    std::string serialize__Vector_to__JSON(){
+        std::stringstream os;
+        {
+            cereal::JSONOutputArchive archive_out(os);
+            serialize(archive_out);
+        }
+        return os.str();
+    }
+
+    // json_string provided must have expected format as created by cereal::JSONOutputArchive.
+    std::vector<float> deserialize__JSON_to__Vector(const std::string & json_string){
+        std::stringstream is (json_string);
+        {
+            cereal::JSONInputArchive archive_in(is);
+            serialize(archive_in);
+        }
+        return _vec_f;
+    }
+    
+    std::vector<float> deserialize__JSON_file_to__Vector(const std::string & filename_string){
+        std::ifstream is(filename_string);
+        {
+            cereal::JSONInputArchive archive_in(is);
+            serialize(archive_in);
+        }
+        return _vec_f;
+    }
+
+    void load_random(int i_vec_size){
+        for(int i = 0; i < i_vec_size; i++) 
+        {
+            float f_rand_value = get_random_float_using_whole_range();
+            _vec_f.push_back(f_rand_value);
```

Comment:
avoid `push_back`. Better resize before the loop and use `[i]` access.
LOOK AT OTHER PLACES WITH THE SAME SITUATION